sycl/source/kernel_bundle.cpp

```diff
@@ -146,7 +146,14 @@ get_empty_interop_kernel_bundle_impl(const context &Ctx,
 
 std::shared_ptr<detail::kernel_bundle_impl>
 join_impl(const std::vector<detail::KernelBundleImplPtr> &Bundles) {
-  return std::make_shared<detail::kernel_bundle_impl>(Bundles);
+  return std::make_shared<detail::kernel_bundle_impl>(Bundles,
+                                                      bundle_state::input);
```

Comment:
Sorry, missed this during previous reviews.
Can we have something like:
```suggestion
                                                      Bundles[0]->get_bundle_state());
```
instead of introducing a new API?

Reply:
That assumes `Bundles` isn't empty. This is really a minor bug-fix included in this PR.